JASP-Engine/JASP/R/binomialtestbayesian.R

```diff
@@ -547,1229 +289,228 @@ BinomialTestBayesian <- function(dataset = NULL, options, perform = "run",
+          p <- JASPgraphs::PlotPriorAndPosterior(dfLines = dfLinesPP, dfPoints = dfPointsPP, xName = xName, BF01 = 1/BF10,
+                                                 CRI = ppCri, median = medianPosterior, drawCRItxt = TRUE, bfSubscripts = bfSubscripts)
+        }
+        else {
+          p <- JASPgraphs::PlotPriorAndPosterior(dfLines = dfLinesPP, dfPoints = dfPointsPP, xName = xName, bfSubscripts = bfSubscripts
+          )
+        }
+        plot <- createJaspPlot(
+          title       = "Prior and Posterior",
+          width       = 530,
+          height      = 400,
+          plot        = p,
+          aspectRatio = 0.7
+        )
+        plot$dependOn(c("plotPriorAndPosterior", "Prior", "hypothesis", "plotPriorAndPosteriorAdditionalInfo"))
+        bayesianBinomPlots[[splitWord]][[paste0(var, lev)]] <- plot
+      }
+      if(options$plotSequentialAnalysis){
+        dfLinesSR   <- .dfLinesSR(d=d, var=var, split = split, a = a, b = b, hyp = hyp, theta0=options$testValue)
+        dfPointsSR  <- NULL
+        xName  <- "n"
+        p <- JASPgraphs::PlotRobustnessSequential(dfLines = dfLinesSR, dfPoints = dfPointsSR, xName = xName, BF01 = 1/BF10, hasRightAxis = TRUE, bfSubscripts = bfSubscripts)
+        plot <- createJaspPlot(
+          title       = "Sequential Analysis",
+          width       = 530,
+          height      = 400,
+          plot        = p,
+          aspectRatio = 0.7
+        )
+        plot$dependOn(c("plotSequentialAnalysis", "Prior", "hypothesis"))
+        bayesianBinomPlots[[splitWord]][[paste0(var, lev, "sequential")]] <- plot
+        
+        
+      }
+    }
+  }
 }
 
-.credibleIntervalPlusMedian <- function(credibleIntervalInterval = .95, a = 1, b = 1, counts = 10, n = 20, hypothesis = "two.sided", theta0 = .5) {
-	
-	lower <- (1 - credibleIntervalInterval) / 2
-	upper <- 1 - lower
-	
-	if (hypothesis == "two.sided") {
-		
-		quantiles <- qbeta(c(lower, .5, upper), a + counts , b + n - counts)
-		
-	} else if (hypothesis == "greater") {
-		
-		rightArea <- pbeta(theta0, a + counts , b + n - counts, lower.tail = FALSE)
-		leftArea <- 1 - rightArea
-		quantiles <- qbeta(leftArea + rightArea * c(lower, .5, upper), a + counts , b + n - counts)
-		
-	} else if (hypothesis == "less") {
-		
-		leftArea <- pbeta(theta0, a + counts , b + n - counts)
-		quantiles <- qbeta(leftArea * c(lower, .5, upper), a + counts , b + n - counts)
-		
-	}
-	
-	return(list(ci.lower = quantiles[1], ci.median = quantiles[2], ci.upper = quantiles[3]))
-	
+.dfLinesPP <- function(dataset, a = 1, b = 1, hyp = "two-sided", theta0=.5, counts, n){
+  if (a == 1 && b == 1) {
+    
+    theta <- seq(0, 1, length.out = 1000)
+    
+  } else {
+    
+    theta <- seq(0.001, 0.999, length.out = 1000)
+  }
+  x <- theta
+  if      (hyp == "two.sided"){
+    linesGroup <- c(dbeta(x, a + counts, b + n - counts), dbeta(x, a, b))
+  }
+  else if (hyp== "greater")   {
+    linesGroup <- c(ifelse (x >= theta0, 
```

Comment:
the `ifelse` statements are a little bit confusing (and inefficient). How about something like:

```r
size <- length(x)
linesPrior <- linesPosterior <- numeric(size) # initializes everything to 0
idx <- x >= theta0 # the nonzero components
# fill in the posterior
linesPosterior[idx] <- dbeta(x[idx], a + counts, b + n - counts) / pbeta(theta0, a + counts, b + n - counts, lower.tail = FALSE)
# fill in the prior
linesPrior[idx] <- dbeta(x[idx], a, b) / pbeta(theta0, a, b, lower.tail = FALSE)
linesGroup <- c(linesPosterior, linesPrior)
```

same goes for the `hyp == "less"` case. Then it becomes very clear that the only difference between `hyp == "greater" and `hyp == "less"` is whether `idx <- x >= theta0` or `idx <- x <= theta0`, and you could simplify the statement.